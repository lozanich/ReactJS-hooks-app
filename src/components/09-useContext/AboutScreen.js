import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const handleLogout = () => {
    setUser({});
  };
  return (
    <div>
      <h1>About Screen</h1>
      <hr />

      <pre>{JSON.stringify(user)}</pre>

      <button onClick={handleLogout} className="btn btn-warning">
        Logout
      </button>
    </div>
  );
};
