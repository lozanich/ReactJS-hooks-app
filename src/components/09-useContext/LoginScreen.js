import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  const user = {
    id: 123838,
    name: "Ivan Lozano",
    email: "lozanich598@gmail.com",
  };
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />

      <button onClick={() => setUser(user)} className="btn btn-primary">
        Login
      </button>
    </div>
  );
};
