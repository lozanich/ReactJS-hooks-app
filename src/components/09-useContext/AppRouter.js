import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { Navbar } from "./Navbar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />

        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen}></Route>
            <Route exact path="/about" component={AboutScreen}></Route>
            <Route exact path="/login" component={LoginScreen}></Route>

            <Route component={HomeScreen}></Route>
          </Switch>
        </div>
      </div>
    </Router>
  );
};
