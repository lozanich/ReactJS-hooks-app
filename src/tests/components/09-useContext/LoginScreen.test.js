import React from "react";
import "@testing-library/jest-dom";
import { UserContext } from "../../../components/09-useContext/UserContext";
import { mount } from "enzyme";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";

describe("Pruebas en el componente <LoginScreen />", () => {
  const setUser = jest.fn();
  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );
  test("Deberia de coincidir con el snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de llamar al metodo setUSer", () => {
    const button = wrapper.find("button");
    button.simulate("click");

    expect(setUser).toHaveBeenCalledTimes(1);
    expect(setUser).toHaveBeenCalledWith({
      id: expect.any(Number),
      name: expect.any(String),
      email: expect.any(String),
    });
  });
});
