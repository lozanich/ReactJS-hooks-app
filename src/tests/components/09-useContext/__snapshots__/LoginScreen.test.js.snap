// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginScreen /> Deberia de coincidir con el snapshot 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
